server/src/core/payment/transaction-service.ts

```diff
@@ -138,7 +138,7 @@ export class Transactions implements TransactionService {
         fromExternal: false,
         from: user._id,
         type: 'refund',
-        provider: this.sendingProvider().name()
+        provider: this.refundsProvider().name()
       };
 
       const trx = await this.sendFundsToUser(user, args);
```

Comment:
This is not sufficient to ensure that the refunds provider is used. the `this.sendFundsToUser()` function still uses the `sendingProvider()` by default. Maybe you could pass an optional argument to `sendFundsToUser()` to make it use a different provider (the refunds provider) when that argument is set.